packages/changed-elements-react/src/api/ChangesTreeDataProvider.ts

```diff
@@ -474,24 +475,44 @@ export class ChangesTreeDataProvider implements ITreeDataProvider {
       updateFunc,
     );
 
-    let targetNodes = await this._getChangedModelOrSubjectNodes(
+    const targetNodes = await this._getChangedModelOrSubjectNodes(
       targetIModel,
       onlyDeleted,
       true,
       updateFunc,
     );
 
     const currentModelIds = new Set(currentNodes.map((node: TreeNodeItem) => node.id));
-    targetNodes = targetNodes.filter((entry: TreeNodeItem) => !currentModelIds.has(entry.id));
-
-    this._models = [...currentNodes, ...targetNodes];
+    const uniqueNodes = targetNodes.filter((entry: TreeNodeItem) => !currentModelIds.has(entry.id));
+    const mergedNodes = this._findAndMergeModelChanges(currentNodes, targetNodes);
+    this._models = [...mergedNodes, ...uniqueNodes];
 
     // Release, as it will be unused after the creation of models
     this._rootEntries = undefined;
 
     return this._models;
   }
 
+  private _findAndMergeModelChanges = (currentNodes: ReadonlyArray<TreeNodeItem>, targetNodes: ReadonlyArray<TreeNodeItem>) => {
+    const currentNodeMap = new Map<string, TreeNodeItem>();
+    currentNodes.forEach((node: TreeNodeItem) => {
+      const nodeCopy = _.cloneDeep(node);
```

Comment:
Please use [`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) thank you

Reply:
I would rather us not do a clone at all. The list of nodes on a model may be quite substantial depending on the diff and frontends already chug and use a lot of memory during version compare to be doing potentially thousands of clones.

Reply:
Yes should be fine. But worried about unforeseen side effects due to the reference to the currentNodes array